Where_Is_My_Wife/Assets/_Main/Scripts/Animation/AnimationController.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using UniRx;
+using UnityEngine;
+using WhereIsMyWife.Managers;
+using Zenject;
+
+namespace WhereIsMyWife.Controllers
+{
+    public class AnimationController : MonoBehaviour
+    {
+        [SerializeField] private Animator _animator;
+
+        [Inject] private IMovementStateEvents _movementStateEvents;
+        [Inject] private IWallHangStateEvents _wallHangStateEvents;
+        [Inject] private IDashStateEvents _dashStateEvents;
+        [Inject] private IPlayerStateIndicator _stateIndicator;
+
+        private Rigidbody2D _rigidbody2D;
+        
+        private const string RUN_ANIMATION_STATE = "walk";
+        private const string IDLE_ANIMATION_STATE = "idle";
+        private const string FALL_ANIMATION_STATE = "fall";
+        private const string JUMP_ANIMATION_STATE = "jump";
+        private const string CROUCH_ANIMATION_STATE = "crouch";
+        private const string WALL_SLIDE_ANIMATION_STATE = "wall_slide";
+        private const string LAND_ANIMATION_STATE = "land";
+        private const string WALL_HIT_ANIMATION_STATE = "wall_hit";
+
+        private string _currentAnimationState = "";
+
+        private void Awake()
+        {
+            _rigidbody2D = GetComponent<Rigidbody2D>();
+        }
+
+        private void Start()
+        {
+            _movementStateEvents.JumpStart.AsUnitObservable().Subscribe(Jump);
+            _movementStateEvents.Run.AsUnitObservable().Subscribe(Run);
+
+            _wallHangStateEvents.StartWallHang.Subscribe(StartWallHang);
+            _wallHangStateEvents.WallJumpStart.AsUnitObservable().Subscribe(Fall);
```

Comment:
Change these observables to `Action` (`jumpStart += Jump`) or to `UnityAction` (`jumpStart .addListener(Jump)`), the former needs desubscription in `OnDestroy`, the latter doesn't.

Reply:
9c0e58e3c9605211b5534a96e3a812d112cae019